SNUTT-2022/SNUTT/Services/SearchService.swift

```diff
@@ -58,11 +63,27 @@ struct SearchService: SearchServiceProtocol {
         let dto = try await searchRepository.fetchTags(quarter: quarter)
         let model = SearchTagList(from: dto)
         appState.search.searchTagList = model
+        guard let recentTagNames = userDefaultsRepository.get([String].self, key: .recentDepartmentTags) else { return }
+        appState.search.pinnedTagList = model?.tagList.filter { $0.type == .department && recentTagNames.contains($0.text) } ?? []
+    }
+
+    private func _saveDepartmentTagsToUserDefaults(from tagList: [SearchTag]) async throws {
+        let departmentTags = tagList.filter { tag in tag.type == .department &&
+            !appState.search.pinnedTagList.contains { $0.text == tag.text }
+        }
+        var updatedTags = departmentTags + appState.search.pinnedTagList
+        if updatedTags.count > 5 {
+            updatedTags = Array(updatedTags.suffix(5))
```

Comment:
```suggestion
            updatedTags = Array(updatedTags.prefix(5))
```

아래가 잘려야하니 `prefix`가 기획상 맞을 듯..?

Reply:
아 헐